### ISSUE TITLE
Add password strenght meter

### ISSUE THREAD

**Issue by igor2x** [pwsafe/pwsafe#1237]
I am looking for example at the following password strength web site: https://passwordmeter.com/
I have typed in password "test" and show "very week" with 4%.
![image](https://github.com/pwsafe/pwsafe/assets/10895030/4efa30e9-2659-41d5-815a-9a422954f20e)

If I generated 100 character password in pwsafe and I get:
![image](https://github.com/pwsafe/pwsafe/assets/10895030/c610bacb-4ce5-4e4f-8a00-0bf97d0d3d71)

I have also seen some of password strength web pages to have estimated time password could be cracked. For example displaying "1 second", "1 day", "1 week", "1 month", "1 year" and "centuries".

May I suggest to add password strength meter into pwsafe? Something like:
![image](https://github.com/pwsafe/pwsafe/assets/10895030/ef41d960-b9b0-41db-b217-ae0a16f0d13e)

And meter could be one of:
![image](https://github.com/pwsafe/pwsafe/assets/10895030/15a48cf0-eb57-4e69-8561-ca4332b17c08)

The idea is to tell user how strong is actually password they are selecting as password and make sure password they choose is really strong.

**Comment by rafaelx** [comment]
I really like this idea.

**Comment by ronys** [comment]
Agree. I was thinking of using https://github.com/tsyrogit/zxcvbn-c
The main challenge is that this will increase the size of the distributable, due to the need for a dictionary.

**Comment by rafaelx** [comment]
The comment of 'CPasswordCharPool::CheckMasterPassword' in 'src/core/PWCharPool.cpp' refers to the calculation of [Shannon Entropy](http://rosettacode.org/wiki/Entropy) to determine password strength. I have included it here only as additional information.
